### PR TITLE
minimal support for freebsd

### PR DESCRIPTION
**Pull request by haampie**
Detects things as

```
freebsd-freebsd14.0-...
```

bootstrapping clingo works

### CODE REVIEW

File: var/spack/repos/builtin/packages/expat/package.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ class Expat(AutotoolsPackage, CMakePackage):
     # `~libbsd`.
     variant(
         "libbsd",
-        default=sys.platform != "darwin" and sys.platform != "win32",
+        default=sys.platform == "linux",
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/expat/package.py:108]: I think this changes the default behavior on Cray

File: lib/spack/spack/operating_systems/freebsd.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import platform as py_platform
+
+from spack.version import Version
+
+from ._operating_system import OperatingSystem
+
+
+class FreeBSD_os(OperatingSystem):
```
Comment (alalazo) [inline lib/spack/spack/operating_systems/freebsd.py]: Can we use:
```suggestion
class FreeBSDOs(OperatingSystem):
```
or anything without an underscore in a class name?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a couple of minor comments. Apart from that LGTM.

File: var/spack/repos/builtin/packages/expat/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/expat/package.py:108]: No, it's `sys.platform` not `spec.platform`

File: lib/spack/spack/operating_systems/freebsd.py (hunk 2 above)
Comment (haampie) [reply lib/spack/spack/operating_systems/freebsd.py]: I found it hard to read, but ok.

**Review by alalazo — APPROVED** [review]

**Pull request merged**